### ISSUE TITLE
Reproducible Crash When Encrypting/Decrypting RSA

### ISSUE THREAD

**Issue by Frizlab** [vapor/crypto#78]
This simple code crashes (segfault).

```swift
import Crypto

let passphrase = "abcdef"

// From https://www.googleapis.com/oauth2/v3/certs
let key = try RSAKey.components(n: "vvAaaSpfr934Qx0ioFiWsopq7UCfLNn0zjYVbq4bvUcGSXU9kowYmQArR7WlIkjk1moffla0UV75QRaQPATva1oD5xQnnW-20haeMWTSsMgUHoN0Np9AD8ffPz-DfMJBOHIo4REL1BFFS33HSZgPl0hxJ-5UScqr4lW1JMy5XGeRho30dnmKTpakU1Oc35hFYKSea_O2SXfmbqiAkWlWkilEzgHq4pzVWiDZe4ZgfMdD4vqkSNrO_PkBFBT1mnBJztQ1h4v1jvUW-zeYYwIcPTaOX-xOTiGH9uQkcNPpe5pBrIZJqR5VNrDl_bJOmvVlhhXZSn4fkxA8kyQcZXGaTw", e: "AQAB")
let encryptedPassphrase = try RSA.encrypt(passphrase, padding: .pkcs1, key: key)
try print(RSA.decrypt(encryptedPassphrase, padding: .pkcs1, key: key))
```

Reproducing 100% with version 3.3.0 (5605334590affd4785a5839806b4504407e054ac) of Crypto. Tested of macOS 10.14.2, with Xcode 10.1, and the standard Swift version for this Xcode version, and on Debian, with Swift 4.2.1.

**Comment by tanner0101** [comment]
Hi @Frizlab, I've opened a PR (#79) including a fix for the crash you described here. However, the code now throws an error when attempting to decrypt the passphrase. AFAIK, this is expected since usually you encrypt with a public key and decrypt with a private key. I just wanted to check with you before changing the test to expect an error. Thanks!

**Comment by Frizlab** [comment]
Yes indeed, throwing is the expected behavior :)

**Comment by tanner0101** [comment]
This is solved in 3.3.1. Thanks again.
